### PR TITLE
Pin page refactor - wallet reducer refactored to store an enrypted se…

### PR DESCRIPTION
**Pull request by CFDevelopment**
- pin page refactored 
- wallet reducer has been refactored to replace unencrypted wallet object
- ethers wallet is serialized/deserialized and then encrypted/decrypted with AES

### CODE REVIEW

File: src/screens/setup/terms/PinPage.js [hunk 1]
```diff
@@ -43,27 +44,39 @@ class PinPage extends Component {
...
     this.navigate(nextScreenToNavigate);
   }
 
   appDecyprtionProcess = (walletName) => {
-    const hash = this.hashInput();
-    if(this.props.hashedPassword === hash) {
-      const hotWalletObj = { wallet: this.props.wallets[0].hdWallet, name: walletName };
+    const plaintext = this.decryptKey();
+    const wallet = JSON.parse(plaintext);
+    const key = 'privateKey';
```
Comment (GregTheGreek) [inline src/screens/setup/terms/PinPage.js]: not necessary.

File: src/screens/setup/terms/PinPage.js [hunk 2]
```diff
@@ -43,27 +44,39 @@ class PinPage extends Component {
...
     this.props.exitSetup(false);
     this.navigate(nextScreenToNavigate);
   }
 
   appDecyprtionProcess = (walletName) => {
-    const hash = this.hashInput();
-    if(this.props.hashedPassword === hash) {
-      const hotWalletObj = { wallet: this.props.wallets[0].hdWallet, name: walletName };
+    const plaintext = this.decryptKey();
+    const wallet = JSON.parse(plaintext);
```
Comment (GregTheGreek) [inline src/screens/setup/terms/PinPage.js]: can convert into one line

File: src/screens/setup/terms/PinPage.js (hunk 1 above)
Comment (CFDevelopment) [reply src/screens/setup/terms/PinPage.js]: @GregTheGreek done

File: src/screens/setup/terms/PinPage.js (hunk 2 above)
Comment (CFDevelopment) [reply src/screens/setup/terms/PinPage.js]: @GregTheGreek done

**Pull request merged**